<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>645</width>
    <height>142</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>5</number>
   </property>
   <property name="topMargin">
    <number>5</number>
   </property>
   <property name="rightMargin">
    <number>5</number>
   </property>
   <property name="bottomMargin">
    <number>5</number>
   </property>
   <property name="spacing">
    <number>3</number>
   </property>
   <item row="0" column="0">
    <widget class="QPushButton" name="pushButton">
     <property name="text">
      <string>HOME</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="2" column="4" colspan="2">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Steps per full revolution</string>
     </property>
    </widget>
   </item>
   <item row="0" column="4">
    <widget class="QSpinBox" name="currentPositionBox">
     <property name="minimum">
      <number>-10000</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QComboBox" name="pinSet">
     <property name="frame">
      <bool>true</bool>
     </property>
     <item>
      <property name="text">
       <string>A+ : CS1, B+ : CS2, A- : CS3, B- : CS4</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>A+ : CS1, A- : CS2, B+ : CS3, B- : CS4</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="QSpinBox" name="totalStepsBox">
     <property name="minimum">
      <number>10</number>
     </property>
     <property name="maximum">
      <number>20000</number>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="6">
    <layout class="QHBoxLayout" name="configLayout">
     <item>
      <widget class="QProgressBar" name="progressBar">
       <property name="value">
        <number>24</number>
       </property>
       <property name="format">
        <string>Motor position at %v</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="2">
    <widget class="QPushButton" name="pushButton_5">
     <property name="text">
      <string>&gt;</string>
     </property>
     <property name="autoRepeat">
      <bool>true</bool>
     </property>
     <property name="autoRepeatInterval">
      <number>20</number>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QPushButton" name="pushButton_2">
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>&lt;</string>
     </property>
     <property name="autoRepeat">
      <bool>true</bool>
     </property>
     <property name="autoRepeatInterval">
      <number>20</number>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="3">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Move To:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="0" column="5">
    <widget class="QCheckBox" name="holdBox">
     <property name="toolTip">
      <string>Hold the stepper motor position. If unchecked, coils are de energised when not moving.</string>
     </property>
     <property name="text">
      <string>Hold Position</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>initialize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>63</x>
     <y>15</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>26</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>stepLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>259</x>
     <y>25</y>
    </hint>
    <hint type="destinationlabel">
     <x>118</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>stepRight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>359</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>251</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>currentPositionBox</sender>
   <signal>editingFinished()</signal>
   <receiver>Dialog</receiver>
   <slot>stepTo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>27</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pinSet</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>Dialog</receiver>
   <slot>setPins(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>230</x>
     <y>141</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>totalStepsBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Dialog</receiver>
   <slot>setTotalSteps(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>339</x>
     <y>167</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>add()</slot>
  <slot>play()</slot>
  <slot>initialize()</slot>
  <slot>stepLeft()</slot>
  <slot>stepRight()</slot>
  <slot>stepTo()</slot>
  <slot>setPins(int)</slot>
  <slot>setTotalSteps(int)</slot>
 </slots>
</ui>
